### PR TITLE
replay: feature flag consumption of duplicate proofs from blockstore

### PR DESCRIPTION
**Pull request by AshwinSekar**
#### Problem
https://github.com/solana-labs/solana/pull/32962 needs to be feature flagged as it could cause a divergence in fork choice.

#### Summary of Changes
Add a feature gate to ensure smooth rollout to 1.17

### CODE REVIEW

**Comment by behzadnouri** [comment]
How about this?
https://github.com/solana-labs/solana/blob/2971e84ec/core/src/replay_stage.rs#L1228

also adding @bw-solana to review since he reviewed original change.

File: core/src/replay_stage.rs [hunk 1]
```diff
@@ -2109,7 +2110,11 @@ impl ReplayStage {
         );
 
         // If we previously marked this slot as duplicate in blockstore, let the state machine know
-        if !duplicate_slots_tracker.contains(&slot) && blockstore.get_duplicate_slot(slot).is_some()
+        if !duplicate_slots_tracker.contains(&slot)
+            && blockstore.get_duplicate_slot(slot).is_some()
```
Comment (bw-solana) [inline core/src/replay_stage.rs:2114]: nit: Is this the most likely check to fail? Could make it first one we check

**Comment by bw-solana** [comment]
> How about this? https://github.com/solana-labs/solana/blob/2971e84ec/core/src/replay_stage.rs#L1228

Yeah, it should be really minor because it's only during startup, but it impacts fork choice

**Review by bw-solana — APPROVED** [review]
LGTM

File: core/src/replay_stage.rs [hunk 2]
```diff
@@ -1227,8 +1228,12 @@ impl ReplayStage {
             let duplicate_slots = blockstore
                 .duplicate_slots_iterator(bank_forks.root_bank().slot())
                 .unwrap();
-            let duplicate_slot_hashes = duplicate_slots
-                .filter_map(|slot| bank_forks.bank_hash(slot).map(|hash| (slot, hash)));
+            let duplicate_slot_hashes = duplicate_slots.filter_map(|slot| {
+                let bank = bank_forks.get(slot)?;
+                bank.feature_set
```
[the remark this answers is not in the document]
Comment (AshwinSekar) [reply core/src/replay_stage.rs:1233]: the problem is that the `duplicate_slots_iterator` could return slots from different epochs if the restart was performed right after the boundary, so I opted to check it for each bank.

**Review by carllin — APPROVED** [review]

**Pull request merged**